include/Spectra/LinAlg/GramSchmidt.h

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2020 Netherlands eScience Center <f.zapata@esciencecenter.nl>
+//
+// This Source Code Form is subject to the terms of the Mozilla
+// Public License v. 2.0. If a copy of the MPL was not distributed
+// with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef SPECTRA_GRAM_SCHIMDT_H
+#define SPECTRA_GRAM_SCHIMDT_H
+
+#include <Eigen/Core>
+
+namespace Spectra {
+
+/// Gram-Schmidt process to orthoganlize a given basis A. Each column correspond
+/// to a vector on the basis.
+/// The start index indicates from what vector to start
+/// the orthogonalization, 0 is the default.
+/// Warnings: Starting the normalization at index n imples that the previous
+/// n-1 vectors are already orthonormal.
+
+template <typename Scalar>
+class Gramschmidt
```

Comment:
maybe rename Gramschmidt to Orthogonalisation and then have the methods corresponding to the QR, MGS, twice enough etc...